hazelcast/src/main/java/com/hazelcast/jet/impl/JetServiceBackend.java

```diff
@@ -379,4 +406,58 @@ public void startScanningForJobs() {
         jobCoordinationService.startScanningForJobs();
     }
 
+    // Store the metadata about the job jar that is uploaded from client side
+    public void storeJobMetaData(JobMetaDataParameterObject parameterObject) {
+        checkIfCanExecuteJar();
+        jobUploadStore.processJobMetaData(parameterObject);
+    }
+
+    // Store a part of job jar that is uploaded
+    public JobMetaDataParameterObject storeJobMultiPart(JobMultiPartParameterObject parameterObject) {
+        JobMetaDataParameterObject result = null;
+        try {
+            result = jobUploadStore.processJobMultipart(parameterObject);
+        } catch (Exception exception) {
+            jobUploadStore.remove(parameterObject.getSessionId());
+            sneakyThrow(exception);
+        }
+        return result;
+    }
+
+    private void checkIfCanExecuteJar() {
```

Comment:
checkResourceUploadEnabled -> method doesn't check executability of the jar, just if it can be uploaded

Reply:
Ok